sql/session.go

```diff
@@ -100,6 +100,10 @@ func NewSession(ctx context.Context, args SessionArgs, e *Executor, remote net.A
 	ctx = log.WithEventLog(ctx, fmt.Sprintf("sql [%s]", args.User), remoteStr)
 	s.context, s.cancel = context.WithCancel(ctx)
 
+	if remoteStr != "" {
+		s.context = log.WithLogTag(s.context, "remote=", remoteStr)
```

Comment:
i'd feel better about this happening before the call to `context.WithCancel`. In fact I think you can move it under the `remote != nil` check above.

Also, how about client instead of "remote"? Seems less ambiguous.